### PR TITLE
ICU-21600 Increase timeout and separate build/test in Cygwin CI

### PR DESCRIPTION
**Pull request by daniel-ju**
This increases the timeout to 3 hours and separates the build/test steps in the Cygwin CI build. This attempts to address the potential timeouts and more clearly see what step is timing out.

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21600
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by jefgen** [comment]
Note: The `intltest` failure in the `Build and Test C++14` pipeline is unrelated to this change, and is currently failing on the `main` branch.

File: .ci-builds/.azure-pipelines.yml [hunk 1]
```diff
@@ -451,8 +451,14 @@ jobs:
...
-        %CYG_ROOT%\\bin\\bash -lc "cd $(cygpath \"$(Build.SourcesDirectory)\") && cd icu4c/source && ./runConfigureICU Cygwin && make check"
-      displayName: 'Build and test'
+        %CYG_ROOT%\\bin\\bash -lc "cd $(cygpath \"$(Build.SourcesDirectory)\") && cd icu4c/source && ./runConfigureICU Cygwin && make"
+      displayName: 'Build ICU'
+      env:
+        CC: gcc
+        CXX: g++
+    - script: |
+        %CYG_ROOT%\\bin\\bash -lc "cd $(cygpath \"$(Build.SourcesDirectory)\") && cd icu4c/source && make check"
+      displayName: 'Run Tests'
```
Comment (jefgen) [inline .ci-builds/.azure-pipelines.yml:461]: nit: Technically this should probably be something like "Build and Run Tests", as I think `make` only builds the ICU library code, and doesn't actually build any of the test code.
So if we wanted to we could use `make test` to build both the library code and the tests, and then run `make check` to execute the tests. :)

Comment (daniel-ju) [reply .ci-builds/.azure-pipelines.yml:461]: Good point, I'll split the test build as well. :)

Comment (jefgen) [reply .ci-builds/.azure-pipelines.yml:461]: Sounds good, thanks!

**Review by jefgen — APPROVED** [review]
lgtm, thanks @daniel-ju !

**Review by erik0686 — APPROVED** [review]

**Review by jefgen — APPROVED** [review]
lgtm, thanks!

**Pull request merged**